### ISSUE TITLE
Deployment failure

### ISSUE THREAD

**Issue by thavel** [Microsoft/openshift-origin#17]
Hello,

I'm struggling with this Openshift deployment template: my deployment ends up with the following failure:
<img width="835" alt="capture d ecran 2017-07-12 a 17 44 21" src="https://user-images.githubusercontent.com/7070725/28126804-e3b2fd96-672a-11e7-8cb3-9f1957658df7.png">
Detailed message:
```
Deployment failed. {
  "status": "Failed",
  "error": {
    "code": "ResourceDeploymentFailure",
    "message": "The resource operation completed with terminal provisioning state 'Failed'.",
    "details": [
      {
        "code": "DeploymentFailed",
        "message": "At least one resource deployment operation failed. Please list deployment operations for details. Please see https://aka.ms/arm-debug for usage details.",
        "details": [
          {
            "code": "Conflict",
            "message": "{\r\n  \"status\": \"Failed\",\r\n  \"error\": {\r\n    \"code\": \"ResourceDeploymentFailure\",\r\n    \"message\": \"The resource operation completed with terminal provisioning state 'Failed'.\",\r\n    \"details\": [\r\n      {\r\n        \"code\": \"VMExtensionProvisioningError\",\r\n        \"message\": \"VM has reported a failure when processing extension 'deployOpenShift'. Error message: \\\"Enable failed: failed to execute command: command terminated with exit status=2\\\".\"\r\n      }\r\n    ]\r\n  }\r\n}"
          }
        ]
      }
    ]
  }
}
```

It seems that my issue is very similar to [this one](https://github.com/Azure/azure-quickstart-templates/issues/3285) (from the old repo).

So, I tried to deploy my cluster with `centos` and `rhel`, to shorten names and prefixes down to 3 characters (but as @haroldwongms mentioned, it should work with names up to 20 characters now). None of my attempts lead to a successful deployment.

I also can't reach the VMs through SSH:
```
ssh -i <my_ssh_deploy_key> <my_admin_user>@<master_node_public_ip>
ssh: connect to host <master_node_public_ip> port 22: Operation timed out
```
And the "connect" button in the portal is disabled for some reason: 
<img width="97" alt="capture d ecran 2017-07-12 a 17 44 54" src="https://user-images.githubusercontent.com/7070725/28127359-905dc606-672c-11e7-9741-e73454261241.png">

I'm using very basic parameters:
```
{
	"": "http://schema.management.azure.com/schemas/2015-01-01/deploymentParameters.json#",
	"contentVersion": "1.0.0.0",
	"parameters": {
		"_artifactsLocation": {
			"value": "https://raw.githubusercontent.com/Microsoft/openshift-origin/master/"
		},
		"osImage": {
			"value": "centos"
		},
		"masterVmSize": {
			"value": "Standard_DS2_v2"
		},
		"infraVmSize": {
			"value": "Standard_DS2_v2"
		},
		"nodeVmSize": {
			"value": "Standard_DS2_v2"
		},
		"openshiftClusterPrefix": {
			"value": "tst"
		},
		"openshiftMasterPublicIpDnsLabel": {
			"value": "tst-master"
		},
		"infraLbPublicIpDnsLabel": {
			"value": "tst-lb"
		},
		"masterInstanceCount": {
			"value": 1
		},
		"infraInstanceCount": {
			"value": 1
		},
		"nodeInstanceCount": {
			"value": 1
		},
		"dataDiskSize": {
			"value": 128
		},
		"adminUsername": {
			"value": "xxxxxx"
		},
		"openshiftPassword": {
			"value": "Pass@word1"
		},
		"sshPublicKey": {
			"value": "ssh-rsa xxxxxx test@test.local"
		},
		"keyVaultResourceGroup": {
			"value": "tst-rg"
		},
		"keyVaultName": {
			"value": "tst-vault"
		},
		"keyVaultSecret": {
			"value": "tst-key"
		},
		"aadClientId": {
			"value": "xxxxxx"
		},
		"aadClientSecret": {
			"value": "xxxxxx"
		},
		"defaultSubDomainType": {
			"value": "xipio"
		},
		"defaultSubDomain": {
			"value": "none"
		}
	}
}
```
(I put some `xxxxxx` to hide some info)

I used the commands from the README, and the following one to generate my SSH key:
```
ssh-keygen -f ./openshift_rsa -t rsa -N ''
```

Do you have any ideas that would help me solve my problem?

Tibo

**Comment by BorisB2015** [comment]
Hi @thavel, 

Your post reminds me that we need a _separate_ troubleshooting section. 

In the readme there is a section on how to ssh into master node (note the non-standard port):

For further troubleshooting, please SSH into your master0 node on port 2200. You will need to be root (sudo su -) and then navigate to the following directory: /var/lib/waagent/custom-script/download

Please take a look at the output stdout, stderr (you will have 2 folders there 1, and 2, for each custom-script file in the template). Look at the latest and see what the error says - it is most likely will be error from ansible playbook which will help guide troubleshooting further. Please post findings here. Thanks!

**Comment by eekwong** [comment]
Check log BorisB2015 suggested.
If you see you have a problem executing 'az' or 'azure', that means the yum install failed 
I have this to bypass it
https://github.com/eekwong/openshift-origin/commit/b55940c7de59a140d938f04b2e953a2ebeffa05b

**Comment by thavel** [comment]
Thanks for your answers!

I've managed to reach the master VM through SSH, as @BorisB2015 suggested.
Here was the failure's logs:
```
TASK [openshift_sanitize_inventory : Ensure that openshift_use_dnsmasq is true] ***
fatal: [xxxxxx-master-0]: FAILED! => {"changed": false, "failed": true, "msg": "Starting in 3.6 openshift_use_dnsmasq must be true or critical features\nwill not function. This also means that NetworkManager must be installed\nenabled and responsible for management of the primary interface."}
fatal: [xxxxxx-infra-0]: FAILED! => {"changed": false, "failed": true, "msg": "Starting in 3.6 openshift_use_dnsmasq must be true or critical features\nwill not function. This also means that NetworkManager must be installed\nenabled and responsible for management of the primary interface."}
fatal: [xxxxxx-node-0]: FAILED! => {"changed": false, "failed": true, "msg": "Starting in 3.6 openshift_use_dnsmasq must be true or critical features\nwill not function. This also means that NetworkManager must be installed\nenabled and responsible for management of the primary interface."}
	to retry, use: --limit @/home/xxxxxx/openshift-ansible/playbooks/byo/config.retry

PLAY RECAP *********************************************************************
localhost                  : ok=8    changed=0    unreachable=0    failed=0
xxxxxx-infra-0              : ok=3    changed=0    unreachable=0    failed=1
xxxxxx-master-0             : ok=4    changed=0    unreachable=0    failed=1
xxxxxx-node-0               : ok=3    changed=0    unreachable=0    failed=1


Failure summary:

  1. Host:     xxxxxx-master-0
     Play:     Initialize host facts
     Task:     openshift_sanitize_inventory : Ensure that openshift_use_dnsmasq is true
     Message:  Starting in 3.6 openshift_use_dnsmasq must be true or critical features
               will not function. This also means that NetworkManager must be installed
               enabled and responsible for management of the primary interface.

  2. Host:     xxxxxx-infra-0
     Play:     Initialize host facts
     Task:     openshift_sanitize_inventory : Ensure that openshift_use_dnsmasq is true
     Message:  Starting in 3.6 openshift_use_dnsmasq must be true or critical features
               will not function. This also means that NetworkManager must be installed
               enabled and responsible for management of the primary interface.

  3. Host:     xxxxxx-node-0
     Play:     Initialize host facts
     Task:     openshift_sanitize_inventory : Ensure that openshift_use_dnsmasq is true
     Message:  Starting in 3.6 openshift_use_dnsmasq must be true or critical features
               will not function. This also means that NetworkManager must be installed
               enabled and responsible for management of the primary interface.
```

I've successfully deployed my cluster using @jreypo fixes from #16.

Again, thanks for your help.
